python-checks/src/main/java/org/sonar/python/checks/cdk/CdkPredicate.java

```diff
@@ -93,6 +94,13 @@ public static Predicate<Expression> isString(Set<String> expectedValues) {
     return expression -> CdkUtils.getString(expression).filter(expectedValues::contains).isPresent();
   }
 
+  /**
+   * @return Predicate which tests if expression is a string matches the pattern
+   */
+  public static Predicate<Expression> isString(Pattern pattern) {
```

Comment:
I suggest to rename the method to `matches` or `match`, then consumers code is more like English sentence and can't be confused with `isString(String expectedValue)`, see in other place:

```
    ExpressionFlow sensitiveAction = getSensitiveExpression(actions, isString(SENSITIVE_ACTIONS));
    ExpressionFlow sensitiveResource = getSensitiveExpression(resources, matches(SENSITIVE_RESOURCE_PATTERN));
```